gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -264,6 +267,9 @@ public Boolean load(String bucketName) {
   // Function that generates downscoped access token.
   private final Function<List<AccessBoundary>, String> downscopedAccessTokenFn;
 
+  // Short username of current logged in user.
+  private String ugiUserName;
```

Comment:
nit: If we are extracting it in constructor can we make it `final`?

Reply:
This class has multiple constructors, if I make it final will have to send this as a parameter in all of them which will require changes in multiple places wherever the constructors are being called.

Reply:
Done